_posts/2024-02-07-#28950.md

```diff
@@ -8,16 +8,50 @@ authors: [instagibbs]
 components: ["rpc"]
 host: ismaelsadeeq
 status: upcoming
-commit:
+commit: bf6fbea9794db5e8d0adce8ff6182574808cea54
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
+- The [`submitpackage RPC`](https://bitcoincore.org/en/doc/26.0.0/rpc/rawtransactions/submitpackage/) RPC method takes `package` as a required argument. The package must consist of a child with its parents, and none of the parents may depend on one another. The package will be validated according to consensus and mempool policy rules. If all transactions pass, they will be accepted to mempool.
+
+- Package transactions submitted to the `submitpackage` RPC successfully are accepted to the node's mempool, but broadcasted to peers individually because [Package relay](https://bitcoinops.org/en/topics/package-relay/) is currently a work in progress. Successful submission does not mean the transactions will propagate throughout the network whenever the transactions need to be relayed as a package.
+
+- `maxfeerate` is an optional parameter for the `sendrawtransaction` and `testmempoolaccept` RPCs. It prevents transactions from being added to mempool and broadcasted to peers if the transaction feerate exceeds `maxfeerate`, avoiding unintentional payment of a high fee rate.
+
+- `maxburnamount` is an optional parameter for the `sendrawtransaction` RPC. It prevents broadcasting transactions with a data carrier `OP_RETURN` output amount that exceeds `maxburnamount` (default value: 0).
+
+- This PR adds optional parameters `maxfeerate` and `maxburnamount` to the `submitpackage` RPC to perform similar sanity checks on package transactions.
+
+- After decoding each package transaction in [`submitpackage`](https://github.com/bitcoin/bitcoin/blob/e69796c79c0aa202087a13ba62d9fbcc1c8754d4/src/rpc/mempool.cpp#L814), this PR adds a check that ensures that if a transaction output is unspendable and the output amount exceeds the `maxburnamount` value, the `submitpackage` function will abort and return immediately.
+
+- The [`ProcessNewPackage`](https://github.com/bitcoin/bitcoin/blob/e69796c79c0aa202087a13ba62d9fbcc1c8754d4/src/validation.h#L286) function validates package transactions against consensus and policy rules. When invoked with `test_accept` set to `false`, all successfully validated transactions are added to the mempool.
+
+- This PR modifies `ProcessNewPackage` to accept a new optional parameter [`max_sane_feerate`](https://github.com/bitcoin-core-review-club/bitcoin/blob/bf6fbea9794db5e8d0adce8ff6182574808cea54/src/validation.cpp#L1658). Whenever a package transaction's individual feerate exceeds the `max_sane_feerate` value, this and the remaining unvalidated transactions will not be accepted into the mempool. 
```

Comment:
Again, it would be helpful to have notes about how max feerate is checked for each RPC. Though optional.

Reply:
You can also link to our review club meeting on 19339 which goes into more detail.

Reply:
Added the notes in latest push thank you